src/agent.ts

```diff
@@ -0,0 +1,98 @@
+import { Finding, HandleTransaction, TransactionEvent, FindingSeverity, FindingType } from "forta-agent";
+import {
+  DEPLOYER_ADDRESS,
+  DEPLOYMENT_TO_ADDRESS,
+  CREATE_AGENT_SIG,
+  UPDATE_AGENT_SIG,
+  DISABLE_AGENT_SIG,
+} from "./constants";
+
+const createFinding = (
+  name: string,
+  description: string,
+  alertId: string,
+  txEvent: TransactionEvent,
+  data: string
+): Finding => {
+  return Finding.fromObject({
+    name,
+    description,
+    alertId,
+    severity: FindingSeverity.Low,
+    type: FindingType.Info,
+    metadata: {
+      txHash: txEvent.hash,
+      from: txEvent.from,
+      to: txEvent.to?.toLowerCase() || "",
+      data,
+    },
+  });
+};
+
+const provideHandleTransaction = (
+  deployerAddress: string,
+  deploymentToAddress: string,
+  createAgentSig: string,
+  updateAgentSig: string,
+  disableAgentSig: string
+): HandleTransaction => {
+  return async (txEvent: TransactionEvent) => {
+    const findings: Finding[] = [];
+
+    if (
+      txEvent.from.toLowerCase() === deployerAddress.toLowerCase() &&
+      txEvent.to &&
+      txEvent.to.toLowerCase() === deploymentToAddress.toLowerCase()
```

Comment:
The first check (whether `from` is the NM deployer) is good, but the second two could lead to the issue I outline in the comment below about there potentially being an intermediary contract that then calls the AgentRegistry.

Then if you implement `filterFunction` properly, it is in _that_ implementation where you can filter for specific callee contracts. 😎